samcli/local/services/localhost_runner.py

```diff
@@ -0,0 +1,124 @@
+"""Base class for all Services that interact with Local Lambda"""
+
+import logging
+import os
+
+from flask import Response
+
+LOG = logging.getLogger(__name__)
+
+
+class LocalhostRunner(object):
+
+    def __init__(self, lambda_runner, port, host, stderr=None):
+        """
+        Creates a BaseService class
+
+        Parameters
+        ----------
+        lambda_runner samcli.commands.local.lib.local_lambda.LocalLambdaRunner
```

Comment:
this is used only to set `is_debugging` flag. Instead, can't you just take a `is_debugging` flag in the constructor and remove lambda_runner from it? That way this class is only about encapsulating Flask. So any regular class can inherit from this one to immediately become a "service". 

In this light, `BaseService` makes sense for class name, or `BaseLocalService`. Subclasses could be `LocalApigwService` and `LocalLambdaService`. I won't nit pick on the names, but this is my two cents to keep them consistent.

Reply:
Good call out. Will update the `lambda_runner` to `is_debugging`.

NAMING IS HARD!!! :) I like your suggestion more. Will update.